### ISSUE TITLE
SoundScrape doesn't gracefully handle 'Preview' tracks

### ISSUE THREAD

**Issue by Insulince** [Miserlou/SoundScrape#66]
Hello,

Firstly, this is a fantastic thing you have created here, a real time saver, keep up the excellent work!

So here is the problem I have ran into, and I have been using soundscrape for about 6 months or so now and have not encountered this before, though I am not familiar with python so perhaps it is an error in my understanding.
When I try downloading Future's tracklist (https://soundcloud.com/futureisnow/tracks) with the command:
`soundscrape futureisnow`
Nothing happens and the console moves on to the next line to await the next command (indicating it believes it completed the previous command). Now I noticed that a larger amount of the tracks are "Previews", however not all of them, and the tracks I am looking for specifically are NOT previews. In fact, they are in a set: https://soundcloud.com/futureisnow/sets/purple-reign. So I tried to do a set download:
`soundscrape https://soundcloud.com/futureisnow/sets/purple-reign`
Yet still I get the same behavior from the console. Any ideas why this may be, or am I just losing my mind here?

Thanks.

**Comment by Insulince** [comment]
Ah just realized something I kept overlooking, after submitting the command I get:
`403 Client Error: Forbidden`
So that's not good. Any info?

**Comment by Miserlou** [comment]
They thanks for reporting this @Insulince .

I think there are two things going on here now - one is that SoundCloud just started new region and content blocking stuff as part of their new monetization strategy, and I don't know how that has/is going to affect SoundScrape yet.

The other thing is a duplicate of https://github.com/Miserlou/SoundScrape/issues/62 - but I haven't been able to investigate yet. Could be related to https://github.com/Miserlou/SoundScrape/issues/1

Thanks for reporting..

**Comment by Miserlou** [comment]
Ah, yeah, this is really ugly. Man, the new SoundCloud sucks, I don't know if there's going to be a way around this one.

**Comment by Miserlou** [comment]
Okay, this is gonna suck but I'll see what I can do. SoundCloud is getting a lot more hostile to scraping, and now their API is returning lies. :(

**Comment by Insulince** [comment]
Yikes! Well I can assure you it is still working to some degree for me, scraping some other artists as we speak, just not Future unfortunately.

Maybe larger artists have access to greater security measures to counteract scraping that smaller artists don't have access too?

**Comment by Miserlou** [comment]
Something like this, I don't know.

Still, it isn't very hard to get, it just means we have to switch to actual scraping rather than using their official API client. For example: https://cf-media.sndcdn.com/ztb7FkrQMUJG.128.mp3?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLW1lZGlhLnNuZGNkbi5jb20venRiN0ZrclFNVUpHLjEyOC5tcDMiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE0NjAwODI1NTF9fX1dfQ__&Signature=k6i4IinkrNYORgsnQxv3lNtHE40JxpPoHggKZo8r60BaACNVIvUPz1gIaTsbdTCleZdaKhOENm4rUgA4n4iuSecwWgqXwSXDfF2wAywi3IZrrO7Ccs-6fMEnjPXNZvqkHv5i0NqFQD-ZT4yjUWzkEK2HAJlp3pHbsNs9M~AFgmDnyEelNWkGhsTvTTkBupTCLM7UY4zIW3lWoTVTu1sY22U~837vmYda~wTr2UAOi9uq4YssEEeJMJt25VztesGp71CNqGZYm~FTBD357X-pUDa~rvLnJxNIu3mLf-PBXlwQv73S6bTsATyznUBfbhJPpgceUeX7TIhffe8m2hiZHg__&Key-Pair-Id=APKAJAGZ7VMH2PFPW6UQ

**Comment by Insulince** [comment]
Woah! Thats awesome, how did you get that? Again I am quite novice at this, the only scraping I have ever done is through your application. I am totally willing to preform manual scraping for now if it will let me get the rest of the set.

**Comment by Miserlou** [comment]
https://api.soundcloud.com/i1/tracks/242403462/streams?client_id=02gUJC0hH2ct1EGOcYXQIzRFU91c72Ea&app_version=1460030896

        item_id = e.message.rsplit('/', 1)[-1].split('.json')[0]
        streams_url = "https://api.soundcloud.com/i1/tracks/%s/streams?client_id=%s&app_version=%s" % (item_id, AGGRESSIVE_CLIENT_ID, APP_VERSION)
        response = requests.get(streams_url).json()
        track_url = response['http_mp3_128_url']
        download_file(track_url, item_id + '.mp3')

Okay, so it's doable, but it's going to take some code reorg to do this properly. Stay tuned.

**Comment by Miserlou** [comment]
Okay, I just pushed 0.23.9 which VERY CRUDELY fixes this. It doesn't do any ID3 tagging or anything like that, it JUST gets the MP3, so you'll have to do the tagging yourself. And I think it only works per track. More fixes coming, but hopefully this can hold you over for right now.

    pip install soundscrape==0.23.9

**Comment by Insulince** [comment]
Alright, so I believe I have upgraded to version 0.23.9, here is what my console says:
`D:\Users\Justin Reusnow\Music\.---Sound Scrape\soundscrape>python -m pip show soundscrape
---
Metadata-Version: 1.1
Name: soundscrape
Version: 0.23.9
Summary: Scrape an artist from SoundCloud
Home-page: https://github.com/Miserlou/SoundScrape
Author: Rich Jones
Author-email: rich@openwatch.net
License: MIT License
Location: c:\users\justin reusnow\appdata\local\programs\python\python35-32\lib\site-packages\soundscrape-0.23.9-py3.5.egg
Requires: args, clint, demjson, fudge, nose, requests, setuptools, simplejson, soundcloud, wheel, mutagen
Classifiers:
  Environment :: Console
  License :: OSI Approved :: Apache Software License
  Operating System :: OS Independent
  Programming Language :: Python
  Programming Language :: Python :: 2.7
  Programming Language :: Python :: 3.3
  Programming Language :: Python :: 3.4
  Programming Language :: Python :: 3.5
  Topic :: Internet :: WWW/HTTP
  Topic :: Internet :: WWW/HTTP :: Dynamic Content
Entry-points:
  [console_scripts]
  soundscrape = soundscrape.soundscrape:main

D:\Users\Justin Reusnow\Music\.---Sound Scrape\soundscrape>`

However it still is not working when I attempt to download anything from Future.
`soundscrape https://soundcloud.com/futureisnow/sets/purple-reign`
Still yields no results. :/

**Comment by Miserlou** [comment]
Yeah, I think you'll have to do it for individual tracks for now, ex https://soundcloud.com/futureisnow/all-right-prod-by-metro-boomin-moon

**Comment by Insulince** [comment]
```
D:\Users\Justin Reusnow\Music\.---Sound Scrape\soundscrape>python soundscrape.py https://soundcloud.com/futureisnow/all-right-prod-by-metro-boomin-moon
'HTTPError' object has no attribute 'message'

D:\Users\Justin Reusnow\Music\.---Sound Scrape\soundscrape>
```

**Comment by Miserlou** [comment]
Dammit. :[. Ok. Thanks for letting me know. Are you on Python 2 or 3?

**Comment by Insulince** [comment]
Python 3.5.1 I believe. Whatever the latest version of 3 is.

**Comment by Miserlou** [comment]
This should all be fixed now. It'll still only download the 30s preview though. :(
